forte/data/data_pack.py

```diff
@@ -499,6 +499,25 @@ def get_payload_data_at(
         """
         return self.get_payload_at(modality, payload_index).cache
 
+    def get_data_store_attribute_idx(
```

Comment:
I don't think so: 
1. this does not looks like a public api suitable at the data pack layer, it is exposing internal implementation details
2. this function simply calls another function with almost the same signature, why don't we just call `self._data_store.get_datastore_attr_idx(type_name, attr_name)`

Reply:
The fact that we need such access such the data store layer from outside is concerning by itself. Instead of exposing this method to the outside, we should provide other methods for stuff like `top.py` to access the data pack data. 

Note our rule of thumb here, by integrating with data store, we do not want to change the interfaces of data pack.

Reply:
Okay I'll keep that in mind